src/mpi/comm/ulfm_impl.c

```diff
@@ -236,7 +235,7 @@ int MPIR_Comm_agree_impl(MPIR_Comm * comm_ptr, int *flag)
     /* Determine both the result of this function (mpi_errno) and the result
      * of flag that will be returned to the user. */
     MPII_Allreduce_group(MPI_IN_PLACE, values, 2, MPI_INT, MPI_BAND, comm_ptr,
-                         new_group_ptr, MPIR_AGREE_TAG, &errflag);
```

Comment:
leak from previous commit

Reply:
Fixed. Also, I squashed a couple of commits to make each commit compilable.